src/system.rs

```diff
@@ -128,6 +130,384 @@ pub struct LoadAverage {
     pub last_pid: PID,
 }
 
+#[derive(Debug, Clone)]
+pub struct CpuTimes {
+    /// Time spent by normal processes executing in user mode;
+    /// on Linux this also includes guest time
+    pub user: u64,
+
+    /// Time spent by niced (prioritized) processes executing in user mode;
+    /// on Linux this also includes guest_nice time
+    pub nice: u64,
+
+    /// Time spent by processes executing in kernel mode
+    pub system: u64,
+
+    /// Time spent doing nothing
+    pub idle: u64,
+
+    /// Time spent waiting for I/O to complete
+    pub iowait: u64,
+
+    /// Time spent for servicing hardware interrupts
+    pub irq: u64,
+
+    /// Time spent for servicing software interrupts
+    pub softirq: u64,
+
+    /// Time spent by other operating systems running in a virtualized environment
+    pub steal: u64,
+
+    /// Time spent running a virtual CPU for guest operating systems
+    /// under the control of the Linux kernel
+    pub guest: u64,
+
+    /// Time spent running a niced guest
+    /// (virtual CPU for guest operating systems
+    /// under the control of the Linux kernel)
+    pub guest_nice: u64,
+}
+
+impl CpuTimes {
+    /// Initialize CpuTimes struct
+    pub fn new(
+        user: u64,
+        nice: u64,
+        system: u64,
+        idle: u64,
+        iowait: u64,
+        irq: u64,
+        softirq: u64,
+        steal: u64,
+        guest: u64,
+        guest_nice: u64,
+    ) -> CpuTimes {
+        CpuTimes {
+            user,
+            nice,
+            system,
+            idle,
+            iowait,
+            irq,
+            softirq,
+            steal,
+            guest,
+            guest_nice,
+        }
+    }
+
+    /// Calculate the total time of CPU utilization.
+    /// guest time and guest_nice time are respectively include
+    /// on user and nice times
+    fn total_time(&self) -> u64 {
+        self.user
+            + self.nice
+            + self.system
+            + self.idle
+            + self.iowait
+            + self.irq
+            + self.softirq
+            + self.steal
+    }
+
+    /// Return the CpuTimesPercent object that contains the detailed
+    /// CPU times percentage of CPU utilization between two instants.
+    fn cpu_percent_since(&self, past_cpu_times: &CpuTimes) -> CpuTimesPercent {
+        if self.total_time() > past_cpu_times.total_time() {
+            let diff_total = (self.total_time() - past_cpu_times.total_time()) as f64;
+            CpuTimesPercent::new(
+                {
```

Comment:
The blocks here seem to repeat multiple times, it may be worth using a function here to avoid repetition and make it easier to read.